app/controllers/collaborations_controller.rb

```diff
@@ -30,10 +34,7 @@ def create
     # if(not @project.assignment_id.nil?)
     #   render plain: "Assignments cannot have collaborators. Please contact admin." and return
     # end
-
-    if(@project.author_id!=current_user.id)
-      render plain: "Access Restricted " and return
-    end
+    authorize @project, :author_access?
```

Comment:
I might have got this wrong, why not have authorize @assignment, :admin_access? as a before action hook? just like edit, update, destroy and reopen.